### DISCUSSION TITLE
passwall2食用与DNS套娃交流

### DISCUSSION THREAD

**Discussion by yunxi993** [Openwrt-Passwall/openwrt-passwall2#97] [Ideas]
# 首次使用passwall2（包括路由器重刷系统）务必更新规则，否则会出现某些设备无法访问某些网站、应用.

### 如果某些应该直连但是走了代理，或者应该代理却走了直连，简单粗暴的方法，例如：
`geosite:bilibili` 加入到Direct
`geosite:pornhub` 加入到Proxy

#

#### 查询国外域名的DNS Server  passwall2 `远程 DNS 协议：UDP-自定义、填入127.0.0.1:端口`，在Direct规则里面填入
`源地址：127.0.0.1`
`端口：你设置的端口`

### 把你插件内设置查询国外域名的DNS加入 Proxy规则
#
# 改动任何外部DNS设置，务必先关闭passwall2，修改完成后在开启passwall2，否则会出现一系列花里胡哨的问题.

# 如果有更好的方案，请分享交流.

**Comment by LitCcc** [comment]
我用的mosdns，配置好mosdns之后在网络-DHCP/DNS-DNS 转发设置127.0.0.1#端口，然后把passwall2 的`直连 DNS 协议` 和 `远程 DNS 协议` 都设为 `UDP`，`直连 DNS` 和 `远程 DNS` 都设为 `127.0.0.1:端口`，这样也可以用

**Reply by yunxi993** [reply]
> 

确实可以，不过之前在我TG群里有讨论过，这样貌似不可以反查
`dig www.baidu.com`
`dig -x IP`
你试试正常吗？

**Reply by LitCcc** [reply]
```
dig www.baidu.com

; <<>> DiG 9.18.1 <<>> www.baidu.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 31431
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;www.baidu.com.                 IN      A

;; ANSWER SECTION:
www.baidu.com.          600     IN      A       220.181.38.150
www.baidu.com.          600     IN      A       220.181.38.149

;; Query time: 3 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Wed Apr 27 13:19:01 CST 2022
;; MSG SIZE  rcvd: 63

dig -x 220.181.38.150

; <<>> DiG 9.18.1 <<>> -x 220.181.38.150
;; global options: +cmd
;; connection timed out; no servers could be reached
```

确实不可以反向解析ip

**Reply by LitCcc** [reply]
试了下你提供的方法，开启passwall2之后同样不可以反查，不知道是哪里设置错了，附上我的设置：
![Snipaste_2022-04-27_14-32-46](https://user-images.githubusercontent.com/36120072/165456027-6de7820a-98b3-4312-a503-c2aaeba6068f.png)
![Snipaste_2022-04-27_14-32-25](https://user-images.githubusercontent.com/36120072/165456035-25052fa1-24b8-4cc1-bb4f-201fd5e74c9e.png)
![Snipaste_2022-04-27_14-32-00](https://user-images.githubusercontent.com/36120072/165456040-19aa5a92-ccec-4627-b9f4-c7403d7e415b.png)


查询国内外域名的DNS已经分别加入Direct和Proxy规则里了

**Reply by yunxi993** [reply]
> 查询国内外域名的DNS已经分别加入Direct和Proxy规则里了

mos里面远程用15353端口，PW2远程随便用个DNS，你在试试.

**Comment by Rashhz** [comment]
其实我感觉直连dns用自动，远程dns用fakedns是最好的了

**Reply by yunxi993** [reply]
> 

没有使用过fakedns，不知道PW2这个有没有修好，之前有issue里面说 fakedns有BUG

**Reply by chancat87** [reply]
> > 
> 
> 没有使用过fakedns，不知道PW2这个有没有修好，之前有issue里面说 fakedns有BUG
不止fakedns ，还有各种莫名的问题
